lib/wasi/src/runtime/resolver/builtin.rs

```diff
@@ -0,0 +1,66 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+use url::Url;
+
+use crate::{
+    bin_factory::BinaryPackage,
+    http::HttpClient,
+    runtime::resolver::{types::ResolverError, types::WebcIdentifier, PackageResolver},
+};
+
+/// The builtin package resolver, backed by WAPM.
+///
+/// Any downloaded assets will be cached on disk.
+#[derive(Debug, Clone)]
+pub struct BuiltinResolver {
+    cache_dir: PathBuf,
+    registry_endpoint: Url,
```

Comment:
Note that supporting multiple registries could be a common use case, eg accessing both dev and prod registry packages. Also need to mix in some local package resolution on top. (for local dev)

Reply:
True. I'm going to leave those more complex use cases for now, though.